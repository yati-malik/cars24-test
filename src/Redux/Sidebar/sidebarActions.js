import { SET_FILTERS } from '../Common/actionTypes';

export function SetFilters(filters){
    return{
        type: SET_FILTERS,
        filters: filters
    }
}