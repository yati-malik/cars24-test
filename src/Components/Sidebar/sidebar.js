import React, { useEffect, useState } from "react";
import { ToastContainer, toast } from "react-toastify";
import { SetFilters as setFilters } from "../../Redux/Sidebar/sidebarActions";
import { connect } from "react-redux";

function Sidebar(props) {

  const [startPrice, setStartPrice] = useState(props.filters.startPrice);
  const [endPrice, setEndPrice] = useState(props.filters.endPrice);

  useEffect(() => {
    setStartPrice(props.filters.startPrice);
    setEndPrice(props.filters.endPrice);
  },[props.filters.startPrice, props.filters.endPrice])

  const handlePriceChange = (e, type) => {
    e.persist();
    let value = e.target.value;
    let reg = /[0-9]/g;
    if(value != '' && !reg.test(value)){
      return false;
    }
    switch (type) {
      case "start":
        setStartPrice(value);
        break;
      case "end":
        setEndPrice(value);
        break;
      default:
    }
  };

  const handleCriteriaApply = () => {
    if (startPrice < endPrice) {
      let criteria = {
        startPrice: startPrice,
        endPrice: endPrice,
      };
      props.setFilters(criteria);
    } else {
      toast.warn("Start price can not be less than end price !", {
        position: toast.POSITION.TOP_LEFT,
      });
      return false;
    }
  };

  return (
    <div className="sidebar">
      <ToastContainer autoClose={3000} />
      <h2>Search Criteria</h2>
      <hr></hr>
      <div>
        <div>
          <span className="bold">Price range</span>
        </div>
        <div>
          <label>Start Price</label>
          <br></br>
          <input
            type="text"
            value={startPrice}
            onChange={(e) => { return handlePriceChange(e, "start")}}
          ></input>
        </div>
        <div>
          <label>End Price</label>
          <br></br>
          <input
            type="text"
            value={endPrice}
            onChange={(e) => {return handlePriceChange(e, "end")}}
          ></input>
        </div>
        <div className="apply">
          <input
            type="button"
            value="Apply"
            onClick={handleCriteriaApply}
          ></input>
        </div>
      </div>
    </div>
  );
}

const mapStateToProps = (state, ownProps) => {
  const { filters } = state.productReducer;
  debugger;
  return { filters };
};

export default connect(mapStateToProps, { setFilters })(Sidebar);
