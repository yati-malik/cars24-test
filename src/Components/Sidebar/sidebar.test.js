import React from "react";
import { configure, mount, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import Sidebar from "./sidebar";
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { render } from "@testing-library/react";

const mockStore = configureStore();
const productReducer = {
    name: '',
    filters: {
        startPrice: 0,
        endPrice: 0
    }
}

configure({ adapter: new Adapter() });

describe('<Sidebar />',() => {
    let sidebar;
    let store;

    beforeEach(() =>{
        store = mockStore({productReducer});
        sidebar = renderer.create(
            <Provider store = {store}>
                <Sidebar />
            </Provider>
        );
    })

    it('renders correctly', () => {
        expect(sidebar.toJSON()).toMatchSnapshot();
    })

});