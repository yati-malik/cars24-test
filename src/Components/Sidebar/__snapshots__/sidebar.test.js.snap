// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sidebar /> renders correctly 1`] = `
<div
  className="sidebar"
>
  <div
    className="Toastify"
  />
  <h2>
    Search Criteria
  </h2>
  <hr />
  <div>
    <div>
      <span
        className="bold"
      >
        Price range
      </span>
    </div>
    <div>
      <label>
        Start Price
      </label>
      <input
        onChange={[Function]}
        type="number"
        value={0}
      />
    </div>
    <div>
      <label>
        End Price
      </label>
      <input
        onChange={[Function]}
        type="number"
        value={0}
      />
    </div>
    <div
      className="apply"
    >
      <input
        onClick={[Function]}
        type="button"
        value="Apply"
      />
    </div>
  </div>
</div>
`;
